src/student/student.service.ts

```diff
@@ -1,4 +1,43 @@
 import { Injectable } from '@nestjs/common';
+import { Student } from "./student.entity";
+import { GetSingleStudentFullDetailsResponse } from "../types";
+import { UpdateStudentDetailsDto } from "./dto/update-student-details.dto";
+import { StudentDegrees } from "./student-degrees.entity";
+import { UpdateStudentDetailsResponse } from "../types/student/update-student-details-response";
 
 @Injectable()
-export class StudentService {}
+export class StudentService {
+
+    async getSingleStudentFullDetails(id: string): Promise<GetSingleStudentFullDetailsResponse> {
+        const student = await Student.findOneOrFail({
+            where: {
+                isActive: true,
+                id,
+            },
+            relations: ['degrees'],
+        })
+        if ( student.degrees !== null ) {
+            delete student.degrees.activationToken;
```

Comment:
here w should delete email and id too becouse we have emial in student table so we dont need get this second time

Reply:
There is no email in table student-degrees . I removed it cause it is redundant -> https://github.com/MichalTomczakSE/Mega-K-Headhunter-Back/pull/9/files#diff-d0fc6b1a0b3b69c5770451a27f1b69b925197b293c54d20f442b990d495ea436L10